lisa/features/resize.py

```diff
@@ -0,0 +1,38 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+from enum import Enum
+from typing import TYPE_CHECKING
+
+from lisa.feature import Feature
+
+FEATURE_NAME_RESIZE = "Resize"
+
+
+class ResizeAction(str, Enum):
+    IncreaseCpuCount = "IncreaseCpuCount"
+    DecreaseCpuCount = "DecreaseCpuCount"
+
+
+class Resize(Feature):
+    @classmethod
+    def name(cls) -> str:
+        return FEATURE_NAME_RESIZE
+
+    @classmethod
+    def can_disable(cls) -> bool:
+        raise NotImplementedError()
+
+    def enabled(self) -> bool:
+        raise NotImplementedError()
+
+    def is_supported(self) -> bool:
+        raise NotImplementedError()
+
+    def resize(self, resize_action: ResizeAction = None) -> None:
+        ...
+
+    def _select_vm_size(
+        self, resize_action: ResizeAction = ResizeAction.IncreaseCpuCount
+    ) -> dict[str, str]:
+        ...
```

Comment:
The private method doesn't need to be declared in the base class, unless it's used in some methods in the base class.